### PR TITLE
Simplify P-Chain transaction creation in unit tests

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

The VM tests pass around a factory that generates builders and signers. I want to remove the explicit usage of builders and signers and replace them all with using the full wallet. Using the full wallet allows modification of the backend to issue multiple dependent transactions without re-creating a builder/signer pair.

Specifically, this PR is working to remove the `txstest.WalletFactory` and `txstest.Backend` types to just use `wallet.Wallet`.

There is also a lot of duplicated code around transaction creation... This PR cleans that up if the code was already touched.

## How this works

Added a new `NewWallet` function. Replaced the usage of `txstest.WalletFactory.NewWallet` with `txstest.NewWallet` in the top-level `platformvm` tests.

## How this was tested

- [X] CI

### CODE REVIEW

File: vms/platformvm/vm_test.go [hunk 1]
```diff
@@ -2074,219 +1983,122 @@ func TestRemovePermissionedValidatorDuringAddPending(t *testing.T) {
...
-		}
-		if castOut.AddressesSet().Equals(set.Of(genesistest.DefaultFundedKeys[1].Address())) {
-			key1OutputAmt += castOut.Amt
-		}
-		if castOut.AddressesSet().Equals(set.Of(changeAddr)) {
-			changeAddrOutputAmt += castOut.Amt
-		}
-		totalOutputAmt += castOut.Amt
-	}
-	require.Equal(totalOutputAmt, key0OutputAmt+key1OutputAmt+changeAddrOutputAmt)
```
Comment (dhrubabasu) [inline vms/platformvm/vm_test.go:2236]: Where are we verifying that the tx actually performed the balance transfer?

Comment (StephenButtolph) [reply vms/platformvm/vm_test.go:2236]: This was never testing that the transfer happened, just that the builder constructed the transaction that we expected.

I don't think this is the place to test that the builder is able to create a valid transaction.

Comment (dhrubabasu) [reply vms/platformvm/vm_test.go:2236]: ... oof

good call on removing this. definitely not the place

**Review by dhrubabasu — APPROVED** [review]

File: vms/platformvm/service_test.go [hunk 2]
```diff
@@ -219,73 +216,70 @@ func TestGetTxStatus(t *testing.T) {
...
 	}
 
 	tests := []test{
 		{
 			"standard block",
-			func(_ *Service, factory *txstest.WalletFactory) (*txs.Tx, error) {
-				builder, signer := factory.NewWallet(testSubnet1ControlKeys[0], testSubnet1ControlKeys[1])
-				utx, err := builder.NewCreateChainTx(
-					testSubnet1.ID(),
+			func(t testing.TB, s *Service) *txs.Tx {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/service_test.go:225]: `t` is passed into `newWallet`, and I felt like it was a bit odd to return an `error` from a function that took in a `testing` value.

File: vms/platformvm/service_test.go [hunk 3]
```diff
@@ -417,7 +405,7 @@ func TestGetBalance(t *testing.T) {
 
 func TestGetStake(t *testing.T) {
 	require := require.New(t)
-	service, _, factory := defaultService(t)
+	service, _ := defaultService(t)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/service_test.go:408]: I think i'd prefer a larger refactor... I am not a fan of these `default...` functions. They are not easily composable or configurable, so they encourage clumsy tests... I'd prefer to leave this to a followup.

File: vms/platformvm/vm_regression_test.go [hunk 4]
```diff
@@ -41,174 +42,120 @@ import (
...
 	vm.ctx.Lock.Lock()
 	defer vm.ctx.Lock.Unlock()
 
+	wallet := newWallet(t, vm, walletConfig{})
+
 	validatorStartTime := vm.clock.Time().Add(executor.SyncBound).Add(1 * time.Second)
 	validatorEndTime := validatorStartTime.Add(360 * 24 * time.Hour)
 
 	nodeID := ids.GenerateTestNodeID()
-	changeAddr := genesistest.DefaultFundedKeys[0].Address()
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/vm_regression_test.go:62]: If the test doesn't care where the change output is going, I don't see a reason to set one explicitly.

File: vms/platformvm/vm_regression_test.go [hunk 5]
```diff
@@ -31,6 +31,7 @@ import (
 	"github.com/ava-labs/avalanchego/utils/constants"
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/vm_regression_test.go:31]: I'm trying to make it easy to write tests - so that then we can write good tests... Admittedly, this file specifically includes edge case testing that at one point (either in a PR, in test, or in prod) was failing unexpected... So they often have a pretty specific set of operations...

File: vms/platformvm/vm_test.go [hunk 6]
```diff
@@ -290,23 +295,27 @@ func TestGenesis(t *testing.T) {
...
 	defer vm.ctx.Lock.Unlock()
 
+	wallet := newWallet(t, vm, walletConfig{})
+
 	var (
-		startTime     = vm.clock.Time().Add(txexecutor.SyncBound).Add(1 * time.Second)
-		endTime       = startTime.Add(defaultMinStakingDuration)
-		nodeID        = ids.GenerateTestNodeID()
-		rewardAddress = ids.GenerateTestShortID()
+		startTime    = vm.clock.Time().Add(txexecutor.SyncBound).Add(1 * time.Second)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/vm_test.go]: Yeah this isn't needed when testing the new-er upgrades. Removed.
